core/modules/file/file.install

```diff
@@ -1254,3 +1254,37 @@ function file_update_1010() {
     $config->save();
   }
 }
+
+/**
+ * Add Apple media types to document file type and to allowed extensions.
+ */
+function file_update_1011() {
+  $config = config('file.type.document');
+  $mimetypes = $config->get('mimetypes');
+  $mimetypes = array_merge($mimetypes, array(
+    'application/vnd.apple.keynote',
+    'application/vnd.apple.numbers',
+    'application/vnd.apple.pages',
+  ));
+  $config->set('mimetypes', $mimetypes);
+  $config->save();
+
+  // Check if there are any existing files with these extensions that are of
+  // type 'undefined'. If so, turn on the reclassification alert.
+  $count = db_query("
+  SELECT COUNT(fid)
+  FROM {file_managed}
+  WHERE 
+  (filename LIKE :keynoteextension OR filename LIKE :numbersextension OR filename LIKE :pagesextension)
+  AND type = :undefinedtype
+  ",
+  array(
+    ':keynoteextension' => '%.keynote',
+    ':numbersextension' => '%.numbers',
+    ':pagesextension' => '%.pages',
+    ':undefinedtype' => 'undefined',
+  ))->fetchField();
```

Comment:
We normally put SQL queries on one line :shrug: and let's separate words so that cspell doesn't complain ever, even in the placeholders:

```suggestion
  $count = db_query("SELECT COUNT(fid) FROM {file_managed} WHERE (filename LIKE :keynote_extension OR filename LIKE :numbers_extension OR filename LIKE :pages_extension) AND type = :undefined_type",
  array(
    ':keynote_extension' => '%.keynote',
    ':numbers_extension' => '%.numbers',
    ':pages_extension' => '%.pages',
    ':undefined_type' => 'undefined',
  ))->fetchField();
```